python/image.bzl

```diff
@@ -72,14 +72,21 @@ def py_image(name, base=None, deps=[], layers=[], **kwargs):
   # a single target can be used for all three.
   native.py_binary(name=binary_name, deps=deps + layers, **kwargs)
 
+  base = base or DEFAULT_BASE
+
+  # Add a separate layer for external deps, usually they don't change very often.
+  if external_layer:
```

Comment:
Making this a single layer will significantly hurt performance when a small dependency is added, changed, or removed.  It also creates an interesting combinatorial problem for cross-image dependency sharing.

I agree that this is a 10-20% case, not an 80-90% case.

For rules_python, I wonder if simply using:
```
py_image(
    name = "foo",
    ...
    layers = all_requirements,
)
```

Would get you what you want?  Maybe, maybe not because IIRC it's intentionally dumb about deduplicating across deps (to keep them agnostic of the binary context).

Reply:
The issue with `layers = all_requirements` is that the `requirements.txt` file in the repo might contain a LOT of pip packages, while only some of them are used in the `py_image` that is being built. I wouldn't want to include the useless packages to the image, and I wouldn't want the built image to change just because somebody adds an unrelated pip package to the `requirements.txt`.